packages/core/crates/core-path/src/channel_graph.rs

```diff
@@ -0,0 +1,144 @@
+use crate::channel_graph::ChannelChange::{CurrentBalance, Epoch, Status};
+use crate::errors::Result;
+use crate::path::Path;
+use core_types::channels::{ChannelEntry, ChannelStatus};
+use core_types::protocol::INTERMEDIATE_HOPS;
+use petgraph::graphmap::DiGraphMap;
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+use utils_log::{debug, info};
+use utils_types::primitives::{Address, Balance};
+
+/// Internal structure that adds additional data to a `ChannelEntry` that
+/// can be used to compute edge weights.
+#[derive(Clone, Copy, Debug, PartialEq, Serialize, Deserialize)]
+struct WeightedChannel {
+    channel: ChannelEntry,
+    weight: f64,
+}
+
+/// Implements a HOPR payment channel graph cached in-memory.
+/// This structure is useful for tracking channel state changes and
+/// packet path finding.
+/// The structure is updated only from the Indexer and therefore contains only
+/// the channels that were *seen* on-chain.
+/// Using this structure is generally faster than querying the DB and therefore
+/// is preferred for per-packet path-finding computations.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ChannelGraph {
+    me: Address,
+    graph: DiGraphMap<Address, WeightedChannel>,
+}
+
+/// Enumerates possible changes on a channel entry update
+#[derive(Clone, Copy, Debug)]
+pub enum ChannelChange {
+    /// Channel status has changed
+    Status { old: ChannelStatus, new: ChannelStatus },
+
+    /// Channel balance has changed
+    CurrentBalance { old: Balance, new: Balance },
+
+    /// Channel epoch has changed
+    Epoch { old: u32, new: u32 },
+}
+
+impl Display for ChannelChange {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Status { old, new } => {
+                write!(f, "Status: {old} -> {new}")
+            }
+
+            CurrentBalance { old, new } => {
+                write!(f, "Balance: {old} -> {new}")
+            }
+
+            Epoch { old, new } => {
+                write!(f, "Epoch: {old} -> {new}")
+            }
+        }
+    }
+}
+
+impl ChannelGraph {
+    /// Maximum number of intermediate hops the automatic path finding algorithm will look for.
+    pub const INTERMEDIATE_HOPS: usize = 3;
```

Comment:
This is actually configurable for the path finding algorithm as it must work for 1, 2 and 3 hops.

Reply:
It is a max value, might rename the variable to `MAX_INTERMEDIATE_HOPS`